[opentelemetry-cpp] build failure (Unknown CMake command "find_dependency")

### Operating system

Ubuntu 22.04

### Compiler

Clang

### Steps to reproduce the behavior

```Shell
`cmake -B build -S test_cpp_project` with a manifest `vcpkg.json`


{
    "name": "testcppproject",
    "version-string": "head",
    "overrides": [],
    "dependencies": [
        {
            "name": "opentelemetry-cpp",
            "version>=": "1.8.3#5",
            "features": [
                "otlp"
            ]
        }
    ]
}


a configuration file `vcpkg-configuration.json`

```json
{
    "default-registry": {
        "kind": "builtin",
        "baseline": "c74f7879a6bafc001f7668163b6db4588787f065"
    },
    "registries": []
}
```

and, inside `CMakeLists.txt`

```cmake
find_package(opentelemetry-cpp CONFIG REQUIRED)
```
```


### Failure logs

```shell
CMake Error at /home/klalumiere/src/coveo/cpp/cmakelibrary/build/vcpkg_installed/x64-linux-dynamic/share/opentelemetry-cpp/opentelemetry-cpp-config.cmake:86 (find_dependency):
  Unknown CMake command "find_dependency".
Call Stack (most recent call first):
  /home/klalumiere/src/vcpkg/scripts/buildsystems/vcpkg.cmake:852 (_find_package)
  TestTarget/CMakeLists.txt:13 (find_package)
```

### Additional context

I understand the error. We add `find_dependency(absl)` in `opentelemetry-cpp-config.cmake.in`, but we forgot `include(CMakeFindDependencyMacro)`. I tested and adding manually `include(CMakeFindDependencyMacro)` mitigates the issue. I'll open a pull request in a few minutes that solves the issue.

--- a/ports/opentelemetry-cpp/add-missing-find-dependency.patch
+++ b/ports/opentelemetry-cpp/add-missing-find-dependency.patch
@@ -1,11 +1,12 @@
 diff --git a/cmake/opentelemetry-cpp-config.cmake.in b/cmake/opentelemetry-cpp-config.cmake.in
-index adae58d..21baab7 100644
+index adae58dd..26427722 100644
 --- a/cmake/opentelemetry-cpp-config.cmake.in
 +++ b/cmake/opentelemetry-cpp-config.cmake.in
-@@ -69,6 +69,7 @@ set(OPENTELEMETRY_VERSION
+@@ -69,6 +69,8 @@ set(OPENTELEMETRY_VERSION
  # ##############################################################################
  
  find_package(Threads)
++include(CMakeFindDependencyMacro)
 +find_dependency(absl)
  
  set_and_check(OPENTELEMETRY_CPP_INCLUDE_DIRS "@PACKAGE_INCLUDE_INSTALL_DIR@")

--- a/ports/opentelemetry-cpp/vcpkg.json
+++ b/ports/opentelemetry-cpp/vcpkg.json
@@ -2,7 +2,7 @@
   "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
   "name": "opentelemetry-cpp",
   "version-semver": "1.8.3",
-  "port-version": 5,
+  "port-version": 6,
   "description": [
     "OpenTelemetry is a collection of tools, APIs, and SDKs.",
     "You use it to instrument, generate, collect, and export telemetry data (metrics, logs, and traces) for analysis in order to understand your software's performance and behavior."

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5926,7 +5926,7 @@
     },
     "opentelemetry-cpp": {
       "baseline": "1.8.3",
-      "port-version": 5
+      "port-version": 6
     },
     "opentracing": {
       "baseline": "1.6.0",

--- a/versions/o-/opentelemetry-cpp.json
+++ b/versions/o-/opentelemetry-cpp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "dfd999f93f24e916631d57bf73c0bf02f8c2da0a",
+      "version-semver": "1.8.3",
+      "port-version": 6
+    },
+    {
       "git-tree": "d01c3006eaa3db92a643c8b6e275e5275561d443",
       "version-semver": "1.8.3",
       "port-version": 5

